pkgs/development/python-modules/generic/default.nix

```diff
@@ -79,86 +68,47 @@ python.stdenv.mkDerivation (builtins.removeAttrs attrs ["disabled"] // {
     runHook preConfigure
 
     # patch python interpreter to write null timestamps when compiling python files
-    # with following var we tell python to activate the patch so that python doesn't
-    # try to update them when we freeze timestamps in nix store
+    # this way python doesn't try to update them when we freeze timestamps in nix store
     export DETERMINISTIC_BUILD=1
 
-    # prepend following line to import setuptools before distutils
-    # this way we make sure setuptools monkeypatches distutils commands
-    # this way setuptools provides extra helpers such as "python setup.py test"
-    sed -i '0,/import distutils/s//import setuptools;import distutils/' setup.py
-    sed -i '0,/from distutils/s//import setuptools;from distutils/' setup.py
-
     runHook postConfigure
   '';
 
-  checkPhase = attrs.checkPhase or ''
-      runHook preCheck
-
-      ${python}/bin/${python.executable} setup.py test
-
-      runHook postCheck
-  '';
-
   buildPhase = attrs.buildPhase or ''
     runHook preBuild
-
-    ${python}/bin/${python.executable} setup.py build ${lib.concatStringsSep " " setupPyBuildFlags}
-
+    cp ${setuppy} nix_run_setup.py
```

Comment:
Why is this `cp` happening? This is a little strange and confusing. Why not just do

```
${python.interpreter} ${./run_setup.py} ${lib.optionalString ....
```

?